File: spec/javascript/test_nav_parent_false.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('NavParentFalse', 'src/nav_parent_codes.bin', function(r) {
  assert.equal(r.childSize, 3);
  assert.equal(r.elementA.foo.code, 73);
  assert.equal(r.elementA.foo.more.toString(), [49, 50, 51].toString());
  assert.equal(r.elementA.bar.foo.code, 66);
  assert.equal(r.elementB.foo.code, 98);
});

File: spec/java/src/io/kaitai/struct/spec/TestNavParentFalse.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.NavParentFalse;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestNavParentFalse extends CommonSpec {
    @Test
    public void testNavParentFalse() throws Exception {
        NavParentFalse r = NavParentFalse.fromFile(SRC_DIR + "nav_parent_codes.bin");

        assertEquals(r.childSize(), 3);
        assertEquals(r.elementA().foo().code(), 73);
        assertEquals(r.elementA().foo().more(), new byte[] { 49, 50, 51 });
        assertEquals(r.elementA().bar().foo().code(), 66);
        assertEquals(r.elementB().foo().code(), 98);
    }
}
